### PR TITLE
Fix Issue #42

### PR DESCRIPTION
**Pull request by NoodleMage**

### CODE REVIEW

**Comment by inorichi** [comment]
I'm against a setting for every viewer. This setting should be either enabled or disabled (globally).

**Comment by inorichi** [comment]
I see a lot of problems with automatically downloading a chapter from the reader. First of all it slows the active chapter (because it starts to download the next one almost at the same time). It also clashes with the preloader, so we're wasting resources (repeated image downloads). I don't see any advantage for this ATM, the preloader is always enabled for the next chapter, so the transition shouldn't be noticeable and it doesn't affect the loading of the active chapter.

**Comment by NoodleMage** [comment]
@inorichi Agree with with automatically downloading. Didn't know it was already pre-loading. Shall remove it from commit then ;). Will also make previous chapter global. later today... ;)

**Comment by j2ghz** [comment]
So no auto download? How about stopping download queue when preloader is downloading? Or making preloader save what it downloads and adding a setting for how much to preload (e.g. 10 chapters)

**Comment by inorichi** [comment]
It adds too much complexity with the current implementation. For something like this, both preloader and downloader should work as one and it would mean a lot of changes, I believe.

And there would be almost no difference between autodownloading and the current preload implementation (unless you want to keep chapters downloaded forever). Next time you open the next chapter, it should continue where it was (unless the disk cache removes them because it needs space).

It's possible, but I think it's not worth the time. Manually using the download next 1/5/10 chapters is already enough IMO.

File: app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsDownloadsFragment.kt [hunk 1]
```diff
@@ -33,6 +34,12 @@ class SettingsDownloadsFragment : SettingsNestedFragment() {
     }
 
     override fun onViewCreated(view: View, savedState: Bundle?) {
+
+        removeAfterReadPref.setOnPreferenceChangeListener { preference, any ->
+            // Enable previous when remove after read is enabled
+            removeAfterReadPreviousPref.isEnabled = !(preferences.removeAfterRead())
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsDownloadsFragment.kt]: Use `android:dependency` for this in the XML

File: app/src/main/res/xml/pref_downloads.xml [hunk 2]
```diff
@@ -3,20 +3,41 @@
...
         android:entryValues="@array/download_slots"
-        android:defaultValue="1"
-        android:summary="%s"/>
+        android:key="@string/pref_download_slots_key"
+        android:summary="%s"
+        android:title="@string/pref_download_slots"/>
+
+    <PreferenceCategory
+        android:persistent="false"
+        android:title="Remove After Read"/>
```
Comment (inorichi) [inline app/src/main/res/xml/pref_downloads.xml]: Make it a string resource, and don't use upper case on every word

**Pull request merged**